src/assets/githubAction/githubActionBeginner.md

```diff
@@ -0,0 +1,125 @@
+
+
+### 4. Steps
+A step is an individual task that can run the commands in a job. These can be actions or commands.
+
+### 5. Actions
+Actions are the standalone commands for the GitHub Actions platform that perform a complex but frequently repeated task. Actions help to reduce the repetitive code in workflows.
+
+### 6. Runner
+A runner is a GitHub actions server where our workflows run. A runner listens for available jobs. Each runner runs multiple jobs at a time, and reports the progress, logs, and results back to GitHub. GitHub hosted runners are based on Ubuntu, Linux, Microsoft Windows, and Mac OS.
+
+Note:
+One event can trigger many workflows, a workflow can contain many jobs, and a job can contain many steps.
+
+### Example:
+
+![process_in_githubActions](/src/assets/githubAction/images/process.png)
+
+### Getting Started:
+
+Before we dive into an example, make sure you have a repository where you want to set up your CI/CD workflow. If you don't have one, create a new repository for example named `action-hero`.
+
+Step 1: Clone your repository into your system
+```bash
+git clone <your_github_repo_url>
+```
+
+Step 2: Go to path `action-hero` and make a folder `.github`
+```bash
+cd action-hero
+mkdir .github
+```
+
+Step 4: Then, go to path `.github` and make a folder `workflows`
+```bash
+cd .github
+mkdir workflows
+```
+
+Step 5: Inside the folder `workflows` create a `hello_world.yml` file
+```bash
+cd workflows
+touch hello_world.yml
+```
+Our file structure will look like this
+```
+📦Action-hero
+┗ 📂.github
+┃ ┣ 📂workflows
+┃ ┃ ┗ 📜hello_world.yml
+```
+
+Add the following code snippet to your `hello_world.yml` file:
+
+```
+name: Hello world workflow
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Print a message
+      run: echo "Hello, GitHub Actions!"
+```
+In this example:
+- The workflow is triggered by a push event on the main branch.
+- The job named "build" runs on the latest version of Ubuntu.
+- The first step checks out the repository using the actions/checkout action.
+- The second step simply prints a message using the echo command.
+
+## GitHub Actions activity:
+
+![alt text](/src/assets/githubAction/images/output.png)
+
+## Conclusion:
+In this blog, we've explored GitHub Actions, an automation and CI/CD service provided by GitHub. We've learned about its key components and how they work together to automate software development workflows.
+
+GitHub Actions enables us to automate tasks like building, testing, and deploying code, leading to improved software quality and time savings.
+
+With a practical example, we've seen how easy it is to set up a basic workflow.
+
+In summary, GitHub Actions is a powerful tool for speedy development processes, making it easier to deliver high-quality software efficiently.
```

Comment:
that's great. If you're planning to write more on this topic you could already add a teaser for your next blog.

Reply:
Currently, I haven't made quite plans for another blog, so  I can't promise a teaser for the next one just yet. But I'll definitely keep your suggestion in mind if I decide to write more on this topic.
 
Would it be fine if I add later?
Or 
Is it not good practice so I need to add it now? @koebel @phil-davis

Reply:
sure, you can update your blog post once you're working on a more advance one on this topic or once you have published another one. 
I agree that it's better not to promise something to the reader that might not happen in the near future.